{"kind":"symbol","topicSections":[{"anchor":"Initializers","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/init(errorCode:context:)"],"title":"Initializers","generated":true},{"anchor":"Instance-Properties","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/context","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/errorCode","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/message"],"title":"Instance Properties","generated":true},{"anchor":"Default-Implementations","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/Error-Implementations"],"title":"Default Implementations","generated":true}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType"},"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore"]]},"abstract":[{"text":"Standard Error type returned by Soto.","type":"text"}],"seeAlsoSections":[{"anchor":"Errors","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/HeaderDecodingError"],"title":"Errors","generated":true}],"variants":[{"paths":["\/documentation\/sotocore\/awserrortype"],"traits":[{"interfaceLanguage":"swift"}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"metadata":{"fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSErrorType","kind":"identifier"}],"role":"symbol","modules":[{"name":"SotoCore"}],"roleHeading":"Protocol","externalID":"s:8SotoCore12AWSErrorTypeP","symbolKind":"protocol","navigatorTitle":[{"kind":"identifier","text":"AWSErrorType"}],"title":"AWSErrorType"},"sections":[],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["Linux"],"tokens":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSErrorType","kind":"identifier"},{"text":" : ","kind":"text"},{"text":"CustomStringConvertible","preciseIdentifier":"s:s23CustomStringConvertibleP","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"Error","preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier"}],"languages":["swift"]}]},{"kind":"content","content":[{"type":"heading","anchor":"overview","text":"Overview","level":2},{"type":"paragraph","inlineContent":[{"text":"Initialized with error code and message.","type":"text"}]}]}],"relationshipsSections":[{"kind":"relationships","title":"Inherits From","identifiers":["doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP","doc:\/\/codes.soto.soto-core\/s5ErrorP","doc:\/\/codes.soto.soto-core\/s8SendableP"],"type":"inheritsFrom"},{"kind":"relationships","title":"Inherited By","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServiceErrorType"],"type":"inheritedBy"},{"kind":"relationships","title":"Conforming Types","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError"],"type":"conformingTypes"}],"references":{"doc://codes.soto.soto-core/documentation/SotoCore/AWSClientError":{"kind":"symbol","abstract":[],"title":"AWSClientError","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSClientError","kind":"identifier"}],"navigatorTitle":[{"text":"AWSClientError","kind":"identifier"}],"role":"symbol","url":"\/documentation\/sotocore\/awsclienterror","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","type":"topic"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType/message":{"type":"topic","title":"message","kind":"symbol","role":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/message","abstract":[],"fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"message","kind":"identifier"},{"text":": ","kind":"text"},{"text":"String","kind":"typeIdentifier","preciseIdentifier":"s:SS"},{"text":"?","kind":"text"}],"url":"\/documentation\/sotocore\/awserrortype\/message"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorContext":{"title":"AWSErrorContext","url":"\/documentation\/sotocore\/awserrorcontext","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","type":"topic","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSErrorContext"}],"navigatorTitle":[{"kind":"identifier","text":"AWSErrorContext"}],"abstract":[{"text":"Additional information about error","type":"text"}],"role":"symbol","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","url":"\/documentation\/sotocore\/awsresponseerror","title":"AWSResponseError","type":"topic","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSResponseError","kind":"identifier"}],"navigatorTitle":[{"text":"AWSResponseError","kind":"identifier"}],"abstract":[{"type":"text","text":"Response Error type returned by Soto if the error code is unrecognised"}],"role":"symbol","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/HeaderDecodingError":{"title":"HeaderDecodingError","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/HeaderDecodingError","url":"\/documentation\/sotocore\/headerdecodingerror","type":"topic","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"HeaderDecodingError","kind":"identifier"}],"navigatorTitle":[{"text":"HeaderDecodingError","kind":"identifier"}],"abstract":[{"text":"Decoding error returned when extracting data out of response headers","type":"text"}],"role":"symbol","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError":{"title":"AWSServerError","url":"\/documentation\/sotocore\/awsservererror","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","type":"topic","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSServerError","kind":"identifier"}],"navigatorTitle":[{"text":"AWSServerError","kind":"identifier"}],"abstract":[],"role":"symbol","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType/context":{"type":"topic","kind":"symbol","title":"context","required":true,"role":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/context","abstract":[{"type":"text","text":"additional context information related to the error"}],"fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"context","kind":"identifier"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:8SotoCore15AWSErrorContextV","text":"AWSErrorContext","kind":"typeIdentifier"},{"text":"?","kind":"text"}],"url":"\/documentation\/sotocore\/awserrortype\/context"},"doc://codes.soto.soto-core/documentation/SotoCore":{"url":"\/documentation\/sotocore","title":"SotoCore","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","type":"topic","abstract":[{"type":"text","text":"The core framework for Soto, a Swift SDK for AWS"}],"role":"collection","kind":"symbol"},"doc://codes.soto.soto-core/s23CustomStringConvertibleP":{"title":"Swift.CustomStringConvertible","type":"unresolvable","identifier":"doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSRawError":{"kind":"symbol","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSRawError","kind":"identifier"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError","url":"\/documentation\/sotocore\/awsrawerror","type":"topic","navigatorTitle":[{"text":"AWSRawError","kind":"identifier"}],"abstract":[{"text":"Raw unprocessed error.","type":"text"}],"role":"symbol","title":"AWSRawError"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType/init(errorCode:context:)":{"url":"\/documentation\/sotocore\/awserrortype\/init(errorcode:context:)","abstract":[{"text":"initialize error","type":"text"}],"role":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/init(errorCode:context:)","kind":"symbol","fragments":[{"kind":"identifier","text":"init"},{"kind":"text","text":"?("},{"kind":"externalParam","text":"errorCode"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:SS","text":"String"},{"kind":"text","text":", "},{"kind":"externalParam","text":"context"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore15AWSErrorContextV","text":"AWSErrorContext"},{"kind":"text","text":")"}],"type":"topic","required":true,"title":"init(errorCode:context:)"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServiceErrorType":{"role":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServiceErrorType","type":"topic","kind":"symbol","url":"\/documentation\/sotocore\/awsserviceerrortype","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSServiceErrorType","kind":"identifier"}],"navigatorTitle":[{"text":"AWSServiceErrorType","kind":"identifier"}],"title":"AWSServiceErrorType","abstract":[{"type":"text","text":"Service that has extended error information"}]},"doc://codes.soto.soto-core/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/codes.soto.soto-core\/s8SendableP"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType":{"title":"AWSErrorType","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","url":"\/documentation\/sotocore\/awserrortype","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSErrorType","kind":"identifier"}],"navigatorTitle":[{"text":"AWSErrorType","kind":"identifier"}],"abstract":[{"type":"text","text":"Standard Error type returned by Soto."}],"role":"symbol","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType/Error-Implementations":{"role":"collectionGroup","title":"Error Implementations","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/Error-Implementations","url":"\/documentation\/sotocore\/awserrortype\/error-implementations","kind":"article","abstract":[]},"doc://codes.soto.soto-core/s5ErrorP":{"title":"Swift.Error","identifier":"doc:\/\/codes.soto.soto-core\/s5ErrorP","type":"unresolvable"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType/errorCode":{"url":"\/documentation\/sotocore\/awserrortype\/errorcode","abstract":[{"text":"Error code return by AWS","type":"text"}],"role":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType\/errorCode","kind":"symbol","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"errorCode"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"type":"topic","required":true,"title":"errorCode"}}}